pkg/router/template/plugin.go

```diff
@@ -155,7 +157,7 @@ func (p *TemplatePlugin) HandleEndpoints(eventType watch.EventType, endpoints *k
 	switch eventType {
 	case watch.Added, watch.Modified:
 		glog.V(4).Infof("Modifying endpoints for %s", key)
-		routerEndpoints := createRouterEndpoints(endpoints, !p.IncludeUDP)
+		routerEndpoints := createRouterEndpoints(endpoints, !p.IncludeUDP, service)
```

Comment:
just to clarify, are you suggesting here that the TemplatePlugin just maintain it's own service listwatch cache (or whatever), and just use that (so, pass a client to the TemplatePlugin constructor, etc)?